### PR TITLE
Auto Increment Port 

### PR DESCRIPTION
**Pull request by al5ina5**
Silently bumps the IP address if taken. Quick fix for several of the reported issues.

### CODE REVIEW

File: lib/reload.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ module.exports = function reload (app, opts, server) {
     }
 
     // Application setup
-    setupClientSideCode()
+    // setupClientSideCode(port)
```
Comment (alallier) [inline lib/reload.js:60]: This is being done below in your changes so please remove the old call here

**Review by alallier — CHANGES_REQUESTED** [review]
Something makes my spidey senses tingle by auto incrementing an explicit port in an application automatically with no choice at all.

I'm inclined to only allow auto incrementing a port on the command line side of reload (which currently has a PR open for #262) but I think there is a possible argument to be made that reload is development only feature and auto incrementing the port in a development environment can make things easier. I'd like to call on @kethinov and @Autre31415 for some second opinions here.

Depending on the result of this conversation the following action items required are:

- [ ] Put the auto increment behind an option and provide a console log stating that an auto increment occurred and what the new port reload is running on
- [ ] Re-run the GitHub Actions and fix errors
- [ ] Write tests

**Comment by kethinov** [comment]
Some thoughts:

1. Auto-incrementing by default seems fine to me. I really see no drawbacks.
2. I agree that it should be a feature that is configurable and can be turned off.
3. The constructor should return the port number that was chosen so the app instantiating reload has that info if they should need it.

**Comment by alallier** [comment]
Closing in favor of #278

**Pull request closed without merging**